transforms/universal/fdedup/spark/src/data_cleaning_transform_spark.py

```diff
@@ -0,0 +1,124 @@
+# (C) Copyright IBM Corp. 2024.
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+import os
+from typing import Any
+
+from data_cleaning_transform import (
+    DataCleaningTransformConfiguration,
+    dataclean_data_access_key,
+    dataclean_data_factory_key,
+    duplicate_list_location_default,
+    duplicate_list_location_key,
+)
+from data_processing.data_access import DataAccessFactoryBase
+from data_processing.transform import TransformStatistics
+from data_processing.utils import get_logger
+from data_processing_spark.runtime.spark import (
+    DefaultSparkTransformRuntime,
+    SparkTransformLauncher,
+    SparkTransformRuntimeConfiguration,
+)
+
+
+logger = get_logger(__name__)
+
+
+class DataCleaningSparkRuntime(DefaultSparkTransformRuntime):
+    """
+    Data cleaning runtime support for Spark
+    """
+
+    def __init__(self, params: dict[str, Any]):
+        super().__init__(params=params)
+        self.logger = get_logger(__name__)
+
+    def get_transform_config(
```

Comment:
Again, this function looks like a duplicate of that in DataCleaningPythonRuntime. Can you make a shared method somehow, either as a global or in a shared superclass?

Reply:
Please refer to the discussion in PR https://github.com/IBM/data-prep-kit/pull/691 as to why it is implemented like this.

Reply:
You seem to be referring to the get_folders() discussion. Can you be more specific about get_transforms_config() and why the cut/paste?

Reply:
nevermind, it does have some ray-specifics.